### PR TITLE
V2 changelog and migration guide

### LINKED ISSUE #28: Add migration guide / changelog
We need to add a migration guide to help our users migrato from v 1.2.x to 2.0

### LINKED ISSUE #33: Update Project's Code examples and Features to comply with V2
Update the Following topics
Explain how to use the project, give some examples for copy/paste
Explain the features itself (client, error handling, request token)

### PR DESCRIPTION
**Pull request by joaohornburg**
Closes #28 
Closes #33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,35 +39,6 @@ Or install it yourself as:
...
-```
-
-### Change Lead Status
-
-```ruby
-rdstation_client = RDStation::Client.new('rdstation_token', 'auth_token')
-rdstation_client.change_lead_status(email: 'joe@foo.bar', status: 'won', value: 999)
-```
-
 ### Authentication
```
Comment (filipenasc) [inline README.md]: In this section, we could also have a "more info" redirecting to developers portal.

File: README.md [hunk 2]
```diff
@@ -139,12 +129,190 @@ contact_info = {
...
+- `RDStation::Error::NotImplemented` (501)
+- `RDStation::Error::BadGateway` (502)
+- `RDStation::Error::ServiceUnavailable` (503)
+- `RDStation::Error::ServerError` (which is returned for 5xx errors different than 500, 501, 502 or 503)
+
+In case of a Bad Request (400), the following speficic errors may be raised (those are subclasses of `RDStation::Error::BadRequest`):
+- `RDStation::Error::ConflictingField`
+- `RDStation::Error::InvalidEventType`
+
+In cause of Unahtorized (401), the following speficic errors may be raised (those are subclasses of `RDStation::Error::Unauthorized`):
```
Comment (filipenasc) [inline README.md]:
```suggestion
In cause of Unauthorized (401), the following speficic errors may be raised (those are subclasses of `RDStation::Error::Unauthorized`):
```

File: README.md [hunk 3]
```diff
@@ -153,14 +321,12 @@ More info: https://developers.rdstation.com/pt-BR/reference/contacts#methodPatch
 4. Push to the branch (`git push origin my-new-feature`)
 5. Create new Pull Request
 
-## Maintainer
-[Nando Sousa](mailto:fernando.sousa@resultadosdigitais.com.br)
+## Maintainers
 
-## Reference
+- [Nando Sousa](mailto:fernando.sousa@resultadosdigitais.com.br)
```
Comment (filipenasc) [inline README.md]: @nandosousafr asked to remove him from this list of maintainers, could you do that?

File: README.md [hunk 4]
```diff
@@ -139,12 +129,190 @@ contact_info = {
...
+  client.webhooks.all
+  client.events.create(my_json_payload)
+  client.fields.all
+```
+
+`RDStation::Contacts`, `RDStation::Events`, `RDStation::Fields` and `RDStation::Webhooks` are not suposed to be instantiated directly anymore. Use `RDStation::Client` to get them instead.
+
+##### Error handling
+
+Now especic errors are raised for each HTTP status:
```
Comment (rafaelhancke) [inline README.md]: Now specific errors are raised for each HTTP status:

File: README.md [hunk 5]
```diff
@@ -139,12 +129,190 @@ contact_info = {
...
+- `RDStation::Error::Conflict` (409)
+- `RDStation::Error::UnsupportedMediaType` (415)
+- `RDStation::Error::UnprocessableEntity` (422)
+- `RDStation::Error::InternalServerError` (500)
+- `RDStation::Error::NotImplemented` (501)
+- `RDStation::Error::BadGateway` (502)
+- `RDStation::Error::ServiceUnavailable` (503)
+- `RDStation::Error::ServerError` (which is returned for 5xx errors different than 500, 501, 502 or 503)
+
+In case of a Bad Request (400), the following speficic errors may be raised (those are subclasses of `RDStation::Error::BadRequest`):
```
Comment (rafaelhancke) [inline README.md]: In case of a Bad Request (400), the following specific errors may be raised

File: README.md (hunk 4 above)
Comment (rafaelhancke) [inline README.md]: Now specific errors are raised for each HTTP status:

File: README.md [hunk 6]
```diff
@@ -139,12 +129,190 @@ contact_info = {
...
+- `RDStation::Error::ConflictingField`
+- `RDStation::Error::InvalidEventType`
+
+In cause of Unahtorized (401), the following speficic errors may be raised (those are subclasses of `RDStation::Error::Unauthorized`):
```
Comment (rafaelhancke) [inline README.md]: In cause of Unauthorized (401), the following speficic errors may be raised

File: README.md [hunk 7]
```diff
@@ -139,12 +129,190 @@ contact_info = {
...
+
+#### Dependencies
+
+`rdstation-ruby-client` now requires `ruby >= 2.0.0`.
+
+### 1.2.1 
+
+#### Deprecations
+
+All API methods that were called directly on `RDStation::Client` (ex: `RDStation::Client.new('rdstation_token', 'auth_token').create_lead(lead_info)`) are now deprecated. Those methos call RDSM's 1.3 API and will be removed in the next release.
```
Comment (rafaelhancke) [inline README.md]: Those methods call

File: README.md [hunk 8]
```diff
@@ -139,12 +131,190 @@ contact_info = {
...
 
 More info: https://developers.rdstation.com/pt-BR/reference/contacts#methodPatchUpsertDetails
 
+### Events
+
+#### Sending a new event
+
+The events endpoint are responsible for receiving different event types in which RD Station Contacts take part in.
+
+It is possible to send default events to RD Station such as conversion events, opportunity marking events and won and lost events. Also, RD Station supports the possibility of receiving different event types, for instance, chat events, ecommerce ones and others.
```
Comment (anacaet) [inline README.md]: It is possible to send default events to RD Station such as conversion events, lifecycle events and won and lost events. Also, RD Station supports the possibility of receiving different event types, for instance, chat events, ecommerce ones and others.

File: README.md [hunk 9]
```diff
@@ -139,12 +131,190 @@ contact_info = {
...
+
+```ruby
+payload = {} # hash representing the payload
+client = RDStation::Client.new(access_token: 'access_token')
+client.events.create(payload)
+```
+
+### Fields
+
+Endpoints to [manage the Fields](https://developers.rdstation.com/en/reference/fields) information in your RD Station account.
```
Comment (anacaet) [inline README.md]: Endpoints to [manage Contact Fields]

**Pull request merged**